jffs2/phoenix-rtos/crc32.c

```diff
@@ -20,10 +20,12 @@
 uint32_t crc32(uint32_t crc, void *p, size_t len)
 {
 	int i;
+	const unsigned char *buf = p;
 	while (len--) {
-		crc ^= *(char *)p++;
-		for (i = 0; i < 8; i++)
+		crc = (crc ^ (*buf++ & 0xff));
```

Comment:
Imho `& 0xff` on unsigned char is kinda too zealous. I guess it's possible for char to be e.g. 9 bits, but it might be 7 bits as well :) Perhaps just declare `buf` as `uint8_t`?